emcc.py

```diff
@@ -1872,7 +1872,17 @@ def default_setting(name, new_default):
     settings.FILESYSTEM = 0
     settings.SYSCALLS_REQUIRE_FILESYSTEM = 0
     settings.JS_LIBRARIES.append((0, 'library_wasmfs.js'))
-    settings.REQUIRED_EXPORTS += ['emscripten_wasmfs_read_file']
+    settings.REQUIRED_EXPORTS += ['_wasmfs_read_file']
+    if settings.FORCE_FILESYSTEM:
+      # Add exports for the JS API. Like the old JS FS, WasmFS by default
+      # includes just what JS parts it actually needs, and FORCE_FILESYSTEM is
+      # required to force all of it to be included if the user wants to use the
+      # JS API directly.
```

Comment:
Do we not have a way to make this more granular today? It would be nice if this worked like native linking, where you get what you use and nothing more without having to specify any special flags. Maybe @sbc100's stub object file idea could help here?

Reply:
Sadly the existing API uses a single JavaScript library object called `FS`, off which all the methods such as `mkdir` hang.     We can only do dependencies and DCE and that JavaScript library entry granularity I think.   So while my idea for better linking works in most JS library cases, I don't think it work in this one :(

Even worse, I'm not sure if any our DCE passes can remove members of the `FS` object, can they?

Reply:
Can we define a more granular new API and provide the legacy `FS` API as an opt-in wrapper?

Reply:
Sure, but I think we should not block this change on such a thing.   I suggest we start with parity as the goal and work on improvements later.

Reply:
To add to that, I think with WasmFS the main usage would be in compiled code anyhow, since that will be most convenient (it's where the API is written and implemented). So I think a simple JS API may be enough. And that API may as well be backwards compatible with the old one.